### PR TITLE
[TR] PIM-4494: Fix js memory leak on a product edit form with scopable attributes

### PR DESCRIPTION
**Pull request by sumbobyboys**
| Q                    | A
| -------------------- | ---
| Bug fix?             | Y
| New feature?         | N
| BC breaks?           | N
| CI currently passes? | Y
| Tests pass?          | Y
| Scenarios pass?      | ?
| Checkstyle issues?*  | N
| PMD issues?**        | N
| Changelog updated?   | Y
| Fixed tickets        |
| Doc PR               |

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/pim-scopable.js [hunk 1]
```diff
@@ -265,30 +259,30 @@ define(
...
 
                 return this;
             },
 
             _reindexFields: function () {
-                this.fields = this.$el.find('[data-scope]');
+                this.fields = _.map(this.$el.find('[data-scope]'), function (field) {
+                    var $field = $(field);
+
+                    return $(field);
```
Comment (jmleroux) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/pim-scopable.js]: `return $field`

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/pim-scopable.js [hunk 2]
```diff
@@ -205,8 +205,8 @@ define(
                 return this;
             },
 
-            _addField: function (field) {
-                this.fieldViews.push(new ScopableField({ el: field }));
+            _addField: function ($field) {
+                this.fieldViews.push(new ScopableField({ el: $field }));
```
Comment (jmleroux) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/pim-scopable.js]: shouldn't it be `this.fieldViews.push(new ScopableField({ $el: $field }));`

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/pim-scopable.js [hunk 3]
```diff
@@ -265,30 +259,30 @@ define(
...
-                        this._setFieldFirst(field);
-                    } else {
-                        $(field).removeClass('first');
-                    }
-                }, this);
+                this._destroyUI();
 
-                this.skipUIInit = false;
-                this._toggle();
+                this._setFieldFirst(this.$el.find('[data-scope]').filter('[data-scope="' + scope + '"]:first'));
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/pim-scopable.js]: `this.$el.find('[data-scope="' + scope + '"]:first')` should work

Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/pim-scopable.js]: yeah, you are right

**Comment by jmleroux** [comment]
GTM

**Comment by phaseinducer** [comment]
GTM

**Pull request merged**